src/tir/schedule/primitive/compute_at.cc

```diff
@@ -160,7 +174,57 @@ int FindInsertionPoint(
   // The valid indices are: (last_producer_position, first_consumer_position]
   ICHECK(split.last_producer_position < split.first_consumer_position);
   // Step 4. Return the last valid insertion point
-  return split.first_consumer_position;
+  int insert_position = split.first_consumer_position;
+  if (index == -1) {
+    return insert_position;
+  } else if (index == -2) {
```

Comment:
The logic for `index == -2` is so confused. Based on the branch `index > 0`, all possible location are always `(last_producer_position, first_consumer_position]`, so can we just return `last_producer_position + 1` in this case?

Reply:
1. For compute at primitive, most of time, we can just return last_producer_position + 1. When other blocks insert ahead of block B, block B maybe far way block C, this is why I add some code to adjust the position. But the adjustment scheme  is not good either. I agree return last_producer_position + 1.
2. For reverse_compute_at, when last_producer_position=-1 we hope it insert the last position, which is mostly caused by parameters.

Reply:
Why do we insert it to the last position in case 2?